src/item.h

```diff
@@ -85,6 +88,96 @@ class item_category
         bool operator!=(const item_category &rhs) const;
 };
 
+/*-----------------------------------------------------------------------------
+ *  Storage for items in items (this is var slim to reduce size to item mem usage)
+ *-----------------------------------------------------------------------------*/
+class storage
+{
+    // save on the clickity clacks
+    typedef std::vector<item>::iterator         item_iterator;
+    typedef std::vector<item>::const_iterator   item_const_iterator;
+
+    private:
+        // <item *> to item that contains <this> storage instance
+        item *me;
+        // items stored in this bag
+        std::vector<item> items;
+        // holds any extra information we need about storage
+        std::map<std::string, void *> var;
```

Comment:
it's a generic type, not solely a void pointer, used for type casted data. prob gonna remove.